src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1076,19 +1074,24 @@ MulticopterPositionControl::update_smooth_takeoff(const float &z_sp, const float
 		}
 
 		// Ramp up takeoff speed.
-		_takeoff_speed += desired_tko_speed * _dt / _takeoff_ramp_time.get();
-		_takeoff_speed = math::min(_takeoff_speed, desired_tko_speed);
+		if (_takeoff_ramp_time.get() > _dt) {
+			_takeoff_speed += desired_tko_speed * _dt / _takeoff_ramp_time.get();
+
+		} else {
+			// No ramp, directly go to desired takeoff speed
+			_takeoff_speed = desired_tko_speed;
+		}
+
+		_takeoff_speed = math::min(_takeoff_speed, _tko_speed.get());
```

Comment:
This prevents the smooth velocity ramp to ramp up correctly. As soon as the desired takeoff setpoint is above the takeoff speed parameter it will get limited and not reach the goal anymore until the desired velocity setpoint comes down again.
The result will be: The ramp will still be faster for high initial desired setpoint but then suddenly gets cut off to the takeoff speed parameter value until you stop vertical movement and start again to go up. Was that intended?